src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -46,7 +46,11 @@ protected sealed override DeclarationModifiers Modifiers
 
         protected void TypeChecks(TypeSymbol type, BindingDiagnosticBag diagnostics)
         {
-            if (type.IsStatic)
+            if (type.IsFileTypeOrUsesFileTypes() && !ContainingType.IsFileTypeOrUsesFileTypes())
```

Comment:
Would it make sense to fold this new accessibility logic into `IsNoMoreVisibleThan`?

Reply:
I don't really think so, even if they are similar concepts. We want to give a different diagnostic than `ERR_BadVis`. Use sites of 'BadVis' error codes do give us a hint as to where we might need to add 'IsFileType' checks, but I think it's still better to have those checks "in parallel" for now.